framework/integration/test/reconciliation/reconciliation_test.go

```diff
@@ -90,20 +91,43 @@ func Test_Finalizer_Integration_Reconciliation(t *testing.T) {
 		t.Fatal("expected", nil, "got", err)
 	}
 
+	// We update the object with a meaningless label to ensure a change in the
+	// ResourceVersion of the object.
+	createdObjAccessor, err := meta.Accessor(createdObj)
```

Comment:
Same idea as in the basic test, we validate that `EnsureCreated` was hit the correct number of times and finalizers are not duplicated.